test/test_ops.py

```diff
@@ -485,6 +390,84 @@ def _distance(a, b):
             msg = "Test passed because the reference was more accurate than the torch operator."
             warnings.warn(msg)
 
+    # Create an input to be tested inside and outside an execution context. For
+    # example, with TorchRefsMode(strict=True) as the context, a function with
+    # torch.foo in it will call refs.foo instead.
+
+    # Setters
+    # Test different inputs inside and outside an execution context.
+    def _set_test_input_different(self, ctx, no_ctx):
+        # keep in sync with get functions
+        return (ctx, no_ctx)
+
+    # Test the same input inside and outside an execution context.
+    def _set_test_input_same(self, x):
+        # keep in sync with get functions
+        return (x, x)
+
+    # Getters
+    def _get_test_input_ctx(self, test_input):
+        # keep in sync with set functions
+        # (ctx, no_ctx)
+        return test_input[0]
+
+    def _get_test_input_no_ctx(self, test_input):
+        # keep in sync with set functions
+        # (ctx, no_ctx)
+        return test_input[1]
+
+    # Create test input functions to check op behavior and API compatibility
+    # inside and outside an execution context. For example, with
+    # TorchRefsMode(strict=True) as the context, a function with torch.foo in it
+    # will call refs.foo instead.
+    #
+    # Note that the torch to refs dispatch must first hit a special conditional
+    # at the start of a torch function before going to the ref. That is why we
+    # have the op_op and op_aliases test input functions. If a torch op and its
+    # ref have incompatible signatures, the call in the said conditional will
+    # fail. The ref_op and ref_op_aliases test input functions exist so that we
+    # could call ref's entrypoint directly.
+
+    # Directly call each ref (in the context) and test it against the
+    # corresponding op (outside of the context). This test calls *different*
+    # entrypoints. Users are not expected to call the ref directly, this is only
+    # done to check for compatibility issues.
+    def _make_test_inputs_ref_op(self, op):  # default
```

Comment:
these ops can be defined inside the tests themselves. This is better because it limits their scope, puts them close to where people may want to find them, and also would not need to take a `self` explicitly, but could just capture them.